jwst/datamodels/flat.py

```diff
@@ -31,5 +31,4 @@ def __init__(self, init=None, **kwargs):
         self.dq = dynamic_mask(self)
 
         # Implicitly create arrays
-        self.dq = self.dq
```

Comment:
What does this change have to do with the filteroffset model?

Reply:
Nothing. I looked at the file because of a naming conflict (fixed since then) and noticed that `self.dq` is defined twice, effectively overwriting the output of `dynamic_mask`. If you prefer I will put it back.

BTW those two statements (claiming to create arrays implicitly)  were removed in `image.py`. I'd like to understand the logic behind these statements and when to keep them.

Reply:
They're really only needed for models where you want to have a dq or err array, but there isn't a dq or err array in the file that's being loaded into the model. This is usually only the case for level-1b science products getting loaded into a `RampModel`. I see no need for it in any of the models for reference files.